ingestion_server/ingestion_server/ingest.py

```diff
@@ -288,9 +289,17 @@ def reload_upstream(
     shared_cols = _get_shared_cols(downstream_db, upstream_db, table)
     upstream_db.close()
 
-    with downstream_db.cursor() as downstream_cur:
-        # Step 2: Create FDW for the data transfer
+    with downstream_db, downstream_db.cursor() as downstream_cur:
+        # Step 2: Create the FDW extension if it does not exist
         log.info("(Re)initializing foreign data wrapper")
+        create_fdw = "CREATE EXTENSION IF NOT EXISTS postgres_fdw;"
+        try:
+            downstream_cur.execute(create_fdw)
+        except psycopg2.errors.UniqueViolation:
+            log.warning("Extension already exists, possible race condition.")
+
+    with downstream_db, downstream_db.cursor() as downstream_cur:
+        # Step 3: Create FDW for the data transfer
```

Comment:
Do you think we could just run `downstream_db.commit()` rather than recreating the cursor context on L301?

Reply:
Using cursor as a context manager automatically makes a transaction of the entire block. Is it expesive to instantiate cursors?

Reply:
Creating a new connection and cursor does come with some overhead, mostly on the Postgres end. Each time a connection is made, [Postgres spins up a new process for that connection](https://www.postgresql.org/docs/13/connect-estab.html). For queries like this that will quickly follow one another, it seems prudent to `commit` with the existing cursor rather than tear down everything and reconstruct it for the second query (IMO). In previous applications I've worked on, we've had long-running processes that issue hundreds of queries open a single connection and create cursors when necessary from there.

Reply:
We're not reconnecting to Postgres, we're reusing the connection created in L282 to just create new cursors, representing individual transactions.

The singular connection created is terminated at the end of the function in L367.

Reply:
Ah, apologies! I thought that the `with downstream_db` was reopening the connection. This is good as is!